### ISSUE TITLE
Errors prevent startup of meteor app with npm dependencies

### ISSUE THREAD

**Issue by serkandurusoy** [phusion/passenger#1582]
I'm on latest passenger nginx integration mode.

I'm trying to run a meteor application (as a meteor app, not a node build) which includes a *local package* which depends on an npm package.

Whenever I try to access it from the browser, I get an error `couldn't read npm version lock information`

To reproduce:

```bash
$ meteor create testapp
$ mkdir -p testapp/packages/test
$ echo -e "test = Npm.require('ilib');\n" > testapp/packages/test/test.js
$ echo -e "Package.describe({name: 'test', version: '0.0.1', summary: 'ilib from npm for test'});\n" > testapp/packages/test/package.js
$ echo -e "Package.onUse(function(api) {\n" >> testapp/packages/test/package.js
$ echo -e "Npm.depends({ilib: '11.0.3'});\n" >> testapp/packages/test/package.js
$ echo -e "api.versionsFrom('1.0');\n" >> testapp/packages/test/package.js
$ echo -e "api.addFiles('test.js', 'server');\n" >> testapp/packages/test/package.js
$ echo -e "api.export('test', 'server');\n" >> testapp/packages/test/package.js
$ echo -e "});\n" >> testapp/packages/test/package.js
$ cd testapp
$ meteor add test 
```

and then create a server in your configuration as a meteor app (not node build) and visit the server address.

You'll get an error displaying `couldn't read npm version lock information` along with other file permission issues.

*Note: I used ilib as an example here, but it is all the same with other npm packages as well.*

**Comment by FooBarWidget** [comment]
I haven't investigated this in detail yet, so I do not yet know what's going on, but can I ask you to deploy a Node build in production? That approach is recommended for production for the reasons listed here: https://www.phusionpassenger.com/library/indepth/meteor/meteor.html

**Comment by serkandurusoy** [comment]
Yes that was my workaround. I also deploy node builds for production.

But the thing is, this is a Q/A server that's being pushed multiple releases every day and we want to keep it aligned with the development environment, hence the meteor deploy.

Node build is fine.

**Comment by FooBarWidget** [comment]
Isn't QA supposed to be aligned with the *production* environment instead? You'd want as much parity with production as possible during testing.

Here's a guide on how to automate deployments of Node builds: https://www.phusionpassenger.com/library/deploy/apache/automating_app_updates/meteor/

**Comment by serkandurusoy** [comment]
Let's name it a *pre-QA* then. We have 3 separate QA environments, 

* One to track upstream development version. Aligned with development so that we don't hit prod/dev differences
* One to track pre-release version. General acceptance testing
* One for release version. For reproducing bug reports

I'm aware of automation options and CI, but my point is not about working around this, for which we alread have done so to generate our node builds. My point is, there is an issue with meteor deployments which is advertised in the documentation as something even the developers would want to do on their machines to be more aligned with production :)

Quoting [from the tutorial](https://github.com/phusion/passenger/wiki/Phusion-Passenger:-Meteor-tutorial#should_use_passenger_in_development)

> **Should Passenger be used in development as well?**
> 
> With all the focus on production, should Passenger be used during development as well? Yes it should!
> 
> Let me be clear: you are in no way obliged to use Passenger in development. You can develop in pure Meteor, then deploy with Passenger. But it is generally a good idea to let your development environment match the production environment as closely as possible. Otherwise you may run into unexpected problems during production.

**Comment by FooBarWidget** [comment]
Yes I understand. I'll have a look at this issue soon.

**Comment by serkandurusoy** [comment]
I just realized that you've published the documentation *library* and I must admit I am quite amazed by how thorough, yet simple it is. Kudos!
